lib/screens/register_screen.dart

```diff
@@ -0,0 +1,298 @@
+                    if (value == null || value.isEmpty) {
+                      return 'Last name cannot be empty';
+                    }
+                    return null;
+                  },
+                  decoration: kTextFieldDecoration.copyWith(
+                      hintText: 'Last Name',
+                      prefixIcon: const Icon(
+                        FontAwesomeIcons.user,
+                        color: kAppColor,
+                      ))),
+              const SizedBox(height: 5.0),
+              TextFormField(
+                autocorrect: false,
+                controller: _bioController,
+                validator: (value) {
+                  if (value == null || value.isEmpty) {
+                    return 'Bio cannot be empty';
+                  }
+                  return null;
+                },
+                decoration: kTextFieldDecoration.copyWith(
+                    hintText: 'Bio',
+                    prefixIcon: const Icon(
+                      FontAwesomeIcons.pen,
+                      color: kAppColor,
+                    )),
+              ),
+              const SizedBox(height: 5.0),
+              TextFormField(
+                  autocorrect: false,
+                  controller: _hometownController,
+                  validator: (value) {
+                    if (value == null || value.isEmpty) {
+                      return 'Hometown cannot be empty';
+                    }
+                    return null;
+                  },
+                  decoration: kTextFieldDecoration.copyWith(
+                      hintText: 'Hometown',
+                      prefixIcon: const Icon(
+                        FontAwesomeIcons.home,
+                        color: kAppColor,
+                      ))),
+              const SizedBox(height: 5.0),
+              TextFormField(
+                  autocorrect: false,
+                  controller: _yearController,
+                  validator: (value) {
+                    if (value == null || value.isEmpty) {
+                      return 'Year cannot be empty';
+                    }
+                    return null;
+                  },
+                  decoration: kTextFieldDecoration.copyWith(
+                      hintText: 'Year of Birth',
+                      prefixIcon: const Icon(
+                        FontAwesomeIcons.birthdayCake,
+                        color: kAppColor,
+                      ))),
+              const SizedBox(height: 5.0),
+              SizedBox(
+                width: double.infinity,
+                child: ElevatedButton(
+                  onPressed: () async {
+                    if (_formKey.currentState!.validate()) {
+                      ScaffoldMessenger.of(context).showSnackBar(
+                          const SnackBar(content: Text('Adding User')));
+                    }
+                    await register();
+                    Navigator.of(context).pushNamed('/login');
```

Comment:
This push is not needed similar to before because when a user is registered in firebase they are automatically logged in. LandingScreen will see this and will push the HomeScreen in response.

Reply:
The stream builder in the landing page that listens to  _auth.userChanges(); does not seem to be updated when the new user is registered.  Unless a different listener to the authentication state can be found that listens to user registration, it makes sense to take the user back to the login page

Reply:
It just needs to be replaced with Navigator.of(context).pop();